EmDash middleware mutates immutable Response headers on Cloudflare/Astro public routes

### Description

`src/astro/middleware.ts` calls `response.headers.set()` directly on the `Response` returned by `next()`.

https://github.com/emdash-cms/emdash/blob/e2f96aa74bd936832a3a4d0636e81f948adb51c7/packages/core/src/astro/middleware.ts#L247-L249

https://github.com/emdash-cms/emdash/blob/e2f96aa74bd936832a3a4d0636e81f948adb51c7/packages/core/src/astro/middleware.ts#L159-L177

On Cloudflare enviroment, some responses which is related to cache api appear to have immutable headers (for example `/_image/*`), which causes: `TypeError: Can't modify immutable headers.`

https://github.com/withastro/astro/blob/7fe40bc7381d981dedad16625d89c00e31cd8fd0/packages/integrations/cloudflare/src/entrypoints/image-transform-endpoint.ts#L8-L14

### Steps to reproduce

using `<Image />` component

astro.config.mjs:

```mjs
import cloudflare from '@astrojs/cloudflare'
import react from '@astrojs/react'
import { d1, r2, sandbox } from '@emdash-cms/cloudflare'
import { defineConfig } from 'astro/config'
import emdash from 'emdash/astro'

export default defineConfig({
	output: 'server',
	adapter: cloudflare({
		imageService: {
			build: 'compile',
                        runtime: 'cloudflare-binding',
		},
	}),
	image: {
		layout: 'constrained',
		responsiveStyles: true,
	},
	integrations: [
		react(),
		emdash({
			database: d1({ binding: 'DB', session: 'auto' }),
			storage: r2({ binding: 'MEDIA' }),
			sandboxRunner: sandbox(),
		}),
	],
	devToolbar: { enabled: false },
})
```

### Environment

- Astro: `6.1.5`
- EmDash: `0.1.1`
- `@astrojs/cloudflare`: `13.1.8`
- `@emdash-cms/cloudflare`: `0.1.1`

### Logs / error output

```shell
01:37:27 [vite] 01:37:26 [200] / 810ms
01:37:28 [vite] 01:37:28 [200] / 114ms
01:37:28 [vite] 01:37:28 [200] /_image 1248ms
01:37:29 [ERROR] [vite] 01:37:29 [ERROR] TypeError: Can't modify immutable headers.
      at setBaselineSecurityHeaders (.../node_modules/.vite/deps_ssr/emdash_middleware.js:1573:20)
      at .../node_modules/.vite/deps_ssr/emdash_middleware.js:1597:7
      at callMiddleware (.../chunk-AJ67EPPI.js:1035:10)
      at _RenderContext.render (.../chunk-AJ67EPPI.js:2079:73)
      at DevApp.render (.../chunk-AJ67EPPI.js:2989:20)
      at Object.handle [as fetch] (.../@astrojs_cloudflare_entrypoints_server.js:208:20)
```

--- a/.changeset/nine-coats-smoke.md
+++ b/.changeset/nine-coats-smoke.md
@@ -1,0 +1,5 @@
+---
+"emdash": patch
+---
+
+Fixes TypeError when setting baseline security headers on Cloudflare responses with immutable headers.

--- a/packages/core/src/astro/middleware.ts
+++ b/packages/core/src/astro/middleware.ts
@@ -160,20 +160,26 @@ async function getRuntime(config: EmDashConfig): Promise<EmDashRuntime> {
  * Baseline security headers applied to all responses.
  * Admin routes get additional headers (strict CSP) from auth middleware.
  */
-function setBaselineSecurityHeaders(response: Response): void {
+function setBaselineSecurityHeaders(response: Response): Response {
+	const headers = new Headers(response.headers);
 	// Prevent MIME type sniffing
-	response.headers.set("X-Content-Type-Options", "nosniff");
+	headers.set("X-Content-Type-Options", "nosniff");
 	// Control referrer information
-	response.headers.set("Referrer-Policy", "strict-origin-when-cross-origin");
+	headers.set("Referrer-Policy", "strict-origin-when-cross-origin");
 	// Restrict access to sensitive browser APIs
-	response.headers.set(
+	headers.set(
 		"Permissions-Policy",
 		"camera=(), microphone=(), geolocation=(), payment=()",
 	);
 	// Prevent clickjacking (non-admin routes; admin CSP uses frame-ancestors)
-	if (!response.headers.has("Content-Security-Policy")) {
-		response.headers.set("X-Frame-Options", "SAMEORIGIN");
+	if (!headers.has("Content-Security-Policy")) {
+		headers.set("X-Frame-Options", "SAMEORIGIN");
 	}
+	return new Response(response.body, {
+		status: response.status,
+		statusText: response.statusText,
+		headers,
+	});
 }
 
 /** Public routes that require the runtime (sitemap, robots.txt, etc.) */
@@ -245,8 +251,7 @@ export const onRequest = defineMiddleware(async (context, next) => {
 			}
 
 			const response = await next();
-			setBaselineSecurityHeaders(response);
-			return response;
+			return setBaselineSecurityHeaders(response);
 		}
 	}
 
@@ -416,8 +421,7 @@ export const onRequest = defineMiddleware(async (context, next) => {
 
 				// Wrap the request in ALS with the per-request db
 				return runWithContext({ editMode: false, db: sessionDb }, async () => {
-					const response = await next();
-					setBaselineSecurityHeaders(response);
+					const response = setBaselineSecurityHeaders(await next());
 
 					// Set bookmark cookie for authenticated users only — they need
 					// read-your-writes consistency across requests. Anonymous visitors
@@ -445,8 +449,7 @@ export const onRequest = defineMiddleware(async (context, next) => {
 		}
 
 		const response = await next();
-		setBaselineSecurityHeaders(response);
-		return response;
+		return setBaselineSecurityHeaders(response);
 	}; // end doInit
 
 	if (playgroundDb) {

--- a/packages/core/src/astro/middleware.ts
+++ b/packages/core/src/astro/middleware.ts
@@ -167,10 +167,7 @@ function setBaselineSecurityHeaders(response: Response): Response {
 	// Control referrer information
 	headers.set("Referrer-Policy", "strict-origin-when-cross-origin");
 	// Restrict access to sensitive browser APIs
-	headers.set(
-		"Permissions-Policy",
-		"camera=(), microphone=(), geolocation=(), payment=()",
-	);
+	headers.set("Permissions-Policy", "camera=(), microphone=(), geolocation=(), payment=()");
 	// Prevent clickjacking (non-admin routes; admin CSP uses frame-ancestors)
 	if (!headers.has("Content-Security-Policy")) {
 		headers.set("X-Frame-Options", "SAMEORIGIN");

--- a/packages/core/src/astro/middleware.ts
+++ b/packages/core/src/astro/middleware.ts
@@ -161,22 +161,14 @@ async function getRuntime(config: EmDashConfig): Promise<EmDashRuntime> {
  * Admin routes get additional headers (strict CSP) from auth middleware.
  */
 function setBaselineSecurityHeaders(response: Response): Response {
-	const headers = new Headers(response.headers);
-	// Prevent MIME type sniffing
-	headers.set("X-Content-Type-Options", "nosniff");
-	// Control referrer information
-	headers.set("Referrer-Policy", "strict-origin-when-cross-origin");
-	// Restrict access to sensitive browser APIs
-	headers.set("Permissions-Policy", "camera=(), microphone=(), geolocation=(), payment=()");
-	// Prevent clickjacking (non-admin routes; admin CSP uses frame-ancestors)
-	if (!headers.has("Content-Security-Policy")) {
-		headers.set("X-Frame-Options", "SAMEORIGIN");
+	const res = new Response(response.body, response);
+	res.headers.set("X-Content-Type-Options", "nosniff");
+	res.headers.set("Referrer-Policy", "strict-origin-when-cross-origin");
+	res.headers.set("Permissions-Policy", "camera=(), microphone=(), geolocation=(), payment=()");
+	if (!res.headers.has("Content-Security-Policy")) {
+		res.headers.set("X-Frame-Options", "SAMEORIGIN");
 	}
-	return new Response(response.body, {
-		status: response.status,
-		statusText: response.statusText,
-		headers,
-	});
+	return res;
 }
 
 /** Public routes that require the runtime (sitemap, robots.txt, etc.) */

